services/QuillLMS/spec/services/vitally_integration/serialize_vitally_sales_user_spec.rb

```diff
@@ -17,6 +17,22 @@
   let!(:student) { create(:user, role: 'student') }
   let!(:old_student) { create(:user, role: 'student') }
 
+  before do
+    previous_year_data = {
+      total_students: 3,
+      active_students: 2,
+      activities_assigned: 3,
+      completed_activities: 3,
+      completed_activities_per_student: 1.5,
+      percent_completed_activities: 1.0,
+      diagnostics_assigned: 2,
+      diagnostics_finished: 2,
+      percent_completed_diagnostics: 1.0
+    }
+    year = Date.today.year - 1
+    $redis.set("teacher_id:#{teacher.id}_vitally_stats_for_year_#{year}", previous_year_data.to_json)
```

Comment:
I think you can utilize your new cache object: `cached_data = CacheVitallyTeacherData.set(teacher.id, year, previous_year_data)`